File: src/client/java/com/somefrills/features/core/PassiveFeature.java
package com.somefrills.features.core;

import io.github.notenoughupdates.moulconfig.observer.Property;

public class PassiveFeature extends AbstractFeature {
    protected PassiveFeature() {
        super(Property.of(false));
    }

    @Override
    protected boolean evaluate() {
        return false;
    }

    @Override
    protected final void onActivate() {
    }

    @Override
    protected final void onDeactivate() {
    }

    @Override
    protected final void onEnable() {
    }

    @Override
    protected final void onDisable() {
    }

}

File: src/main/kotlin/com/somefrills/features/core/PassiveFeature.kt
package com.somefrills.features.core

import io.github.notenoughupdates.moulconfig.observer.Property

open class PassiveFeature protected constructor() : AbstractFeature(Property.of<Boolean>(false)) {
    override fun evaluate(): Boolean {
        return false
    }

    override fun onActivate() {
    }

    override fun onDeactivate() {
    }

    override fun onEnable() {
    }

    override fun onDisable() {
    }
}
